teleport_17: 17.4.2 -> 17.4.5; teleport_16: 16.5.0 -> 16.5.3

JuliusFreudenberger opened the pull request:
Changelog: https://github.com/gravitational/teleport/releases/tag/v17.4.3
Diff: https://github.com/gravitational/teleport/compare/v17.4.2...v17.4.3

Changelog: https://github.com/gravitational/teleport/releases/tag/v16.5.1
Diff: https://github.com/gravitational/teleport/compare/v16.5.0...v16.5.1

I decided to keep everything in one commit, as there are some changes in the upstream folder structure. If I have to split the updates, I would have to temporarily introduce if-else-branches for the different builds.

The folder structure change also forced me to pull a patch in-tree, so I could adapt it.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

JuliusFreudenberger committed:
teleport_17: 17.4.2 -> 17.4.3; teleport_16: 16.5.0 -> 16.5.1

Changelog: https://github.com/gravitational/teleport/releases/tag/v17.4.3
Diff: https://github.com/gravitational/teleport/compare/v17.4.2...v17.4.3

Changelog: https://github.com/gravitational/teleport/releases/tag/v16.5.1
Diff: https://github.com/gravitational/teleport/compare/v16.5.0...v16.5.1

techknowlogick commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 397809`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>teleport (teleport_17)</li>
    <li>teleport.client (teleport_17.client)</li>
    <li>teleport_16</li>
    <li>teleport_16.client</li>
  </ul>
</details>

JuliusFreudenberger committed:
teleport_17: 17.4.3 -> 17.4.5

Changelogs: https://github.com/gravitational/teleport/releases/tag/v17.4.4 https://github.com/gravitational/teleport/releases/tag/v17.4.5
Diff: https://github.com/gravitational/teleport/compare/v17.4.3...v17.4.5

JuliusFreudenberger committed:
teleport_16: 16.5.1 -> 16.5.3

Changelogs: https://github.com/gravitational/teleport/releases/tag/v16.5.1 https://github.com/gravitational/teleport/releases/tag/v16.5.3
Diff: https://github.com/gravitational/teleport/compare/v16.5.1...v16.5.3

arianvp commented:
Note that you can self-merge these with @NixOS/nixpkgs-merge-bot

JuliusFreudenberger commented:
I can? I am not a committer and this is not in by-name, yet.

arianvp commented:
Ah! didn't realize this wasn't in by-name yet

JuliusFreudenberger commented:
Yes, I did consider it but struggled with the Darwin SDK imports and the fact that we have several versions at the same time.
